File: src/main/java/io/eliez/fintools/bin2brand/RangePattern.java
package io.eliez.fintools.bin2brand;

import lombok.Getter;
import lombok.experimental.Accessors;

import java.util.regex.Pattern;

@Getter
@Accessors(fluent = true)
public class RangePattern {
    public static final RangePattern NUMERIC = new RangePattern("\\d+");

    private final Pattern pattern;

    public RangePattern(String boundPattern) {
        this.pattern = Pattern.compile(String.format("(%s)(-(%s))?", boundPattern, boundPattern));
    }
}

File: src/main/kotlin/io/eliez/fintools/bin2brand/RangePattern.kt
package io.eliez.fintools.bin2brand

import java.util.regex.Pattern

class RangePattern(boundPattern: String) {

    val pattern: Pattern

    init {
        this.pattern = Pattern.compile(String.format("(%s)(-(%s))?", boundPattern, boundPattern))
    }

    companion object {
        val NUMERIC = RangePattern("\\d+")
    }
}
